api/v1/storagecluster_types.go

```diff
@@ -137,11 +148,42 @@ type ManageCephObjectStoreUsers struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ExternalStorageMode specifies a mode of the external storage
+type ExternalStorageMode string
+
+const (
+	// ModeOcsToOcs specifies a "ocs-to-ocs" mode of the external storage
+	ModeOcsToOcs ExternalStorageMode = "ocs-to-ocs"
+
+	// ModeStandaloneCeph specifies a "standalone-ceph" mode of the external storage
+	ModeStandaloneCeph ExternalStorageMode = "standalone-ceph"
+)
+
 // ExternalStorageClusterSpec defines the spec of the external Storage Cluster
-// to be connected to the local cluster
+// to be connected to the external cluster
 type ExternalStorageClusterSpec struct {
-	// +optional
+	//+kubebuilder:default:=true
+	// Enable enables the external mode
 	Enable bool `json:"enable,omitempty"`
+
+	//+kubebuilder:default:=standalone-ceph
+	// Mode describes the mode of the external storage
+	Mode ExternalStorageMode `json:"mode,omitempty"`
```

Comment:
Why `Mode`, this is a very generic word? can we try to be more specific?  `ExternalStorage` is by definition defining external mode and we do not want to add another new mode but extend the current one.

Can we rename it to something like `StorageProviderKind`, with 2 possible values: `ocs` and `rhcs` ?

Reply:
would this imply that existing customers have to update `StorageProviderKind` to `rhcs` or the upgrades might break?

Reply:
How about we don't touch the existing External Storage API at all. Something like: 

```
 type StorageClusterSpec struct {
	ExternalStorage ExternalStorageClusterSpec `json:"externalStorage,omitempty"`
	ExternalOCS ExternalOCSClusterSpec
	...
}

type ExternalOCSClusterSpec struct {
	//+kubebuilder:default:=true
	// Enable enables the external mode
	Enable bool `json:"enable,omitempty"`

	// ConnectionString describes the endpoint and UID of the provider
	// server to connect to the API server
	ConnectionString string `json:"connectionString,omitempty"`

	//+kubebuilder:default:="1TB"
	// RequestedCapacity defines the requested storage on the provider
	RequestedCapacity string `json:"requestedCapacity,omitempty"`
}```

`ExternalStorage` would mean we are referring to `rhcs`
`ExternalOCS` would mean we are referring to an external ocs cluster. Its spec `ExternalOCSClusterSpec` would have API connection details and other fields we need to update, like `RequestedCapacity`

Reply:
First, we can decide that the `StorageProviderKind` is defaulting to RHCS which means that everything is backward compatible. 

Second, I think that adding a new section will do more harm than good, as you then need to protect against cases when you have both and the API itself is becoming very confusing from an API user standpoint (like the obvious question, why do I have 2 sections? which one should I use?)

Reply:
done